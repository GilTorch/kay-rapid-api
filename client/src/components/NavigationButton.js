import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import styled from 'styled-components';

const NavButtonStyles=styled.button`
  background:transparent;
  cursor: pointer;
  color:${props => props.highlighted ? "#ff9900ff" : "#606060ff" };
  border:none;
  span{
      display:block;
      font-size:1rem;
  }
  svg {
      font-size:1.5rem;
  }
`

const NavigationButton=({name,iconName,highlighted})=>{
    return(
        <NavButtonStyles highlighted={highlighted} >
            <FontAwesomeIcon icon={iconName}/>
            <span>{name}</span>
        </NavButtonStyles>
    )
}

export default NavigationButton;