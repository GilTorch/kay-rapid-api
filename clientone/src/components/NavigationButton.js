import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import Favorites from '../svg/navigation/favorites-icon.svg';
import FavoritesHighlighted from '../svg/navigation/favorites-icon-highlighted.svg';
import Home from '../svg/navigation/home-icon.svg';
import HomeHighlighted from '../svg/navigation/home-icon-highlighted.svg';
import Profile from '../svg/navigation/profile-icon.svg';
import ProfileHighlighted from '../svg/navigation/profile-icon-highlighted.svg';

const icons={
    "normal":{
        "favorites":Favorites,
        "home":Home,
        "profile":Profile
    },
    "highlighted":{
        "favorites":FavoritesHighlighted,
        "home":HomeHighlighted,
        "profile":ProfileHighlighted
    }
}

const NavigationButton=({name,iconName,highlighted})=>{
    return(
        <button className={highlighted?iconName+"-icon--hightlighted navigation-button--highlighted":"navigation-button"}>
            <img src={highlighted?icons["highlighted"][iconName]:icons["normal"][iconName]} alt="button"/>
            <span className="navigation-button__title">{name}</span>
        </button>
    )
}

export default NavigationButton;