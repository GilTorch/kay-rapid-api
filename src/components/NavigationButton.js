import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

const NavigationButton=({name,iconName})=>{
    return(
        <button className="navigation-button">
            <FontAwesomeIcon icon={iconName} size="2x"/>
            <span className="navigation-button__title">{name}</span>
        </button>
    )
}

export default NavigationButton;