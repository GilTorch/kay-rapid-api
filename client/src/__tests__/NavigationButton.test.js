import React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import 'jest-styled-components';
import NavigationButton from '../components/NavigationButton';

describe('<NavigationButton/>',()=>{
    it('renders and matches snapshot',()=>{
        const wrapper = mount(
                <NavigationButton name="Favori" iconName="heart" highlighted={false}/>
        );
        expect(toJSON(wrapper)).toMatchSnapshot();
    })

    it('has the proper color when highlighted props set to false', ()=>{
        const wrapper = mount(
            <NavigationButton name="Favori" iconName="heart" highlighted={false}/>
        );
        expect(wrapper).toHaveStyleRule('color', '#606060ff');
    })

    it('has the proper color when highlighted props set to true', ()=>{
        const wrapper = mount(
            <NavigationButton name="Favori" iconName="heart" highlighted={true}/>
        );
        expect(wrapper).toHaveStyleRule('color', '#ff9900ff');
    })
})