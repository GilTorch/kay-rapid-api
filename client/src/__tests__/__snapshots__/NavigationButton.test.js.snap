// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationButton/> renders and matches snapshot 1`] = `
.c0 {
  background: transparent;
  cursor: pointer;
  color: #606060ff;
  border: none;
}

.c0 span {
  display: block;
  font-size: 1rem;
}

.c0 svg {
  font-size: 1.5rem;
}

<NavigationButton
  highlighted={false}
  iconName="heart"
  name="Favori"
>
  <styled.button
    highlighted={false}
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  background:transparent;
  cursor: pointer;
  color:",
              [Function],
              ";
  border:none;
  span{
      display:block;
      font-size:1rem;
  }
  svg {
      font-size:1.5rem;
  }
",
            ],
          },
          "displayName": "styled.button",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "button",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      highlighted={false}
    >
      <button
        className="c0"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon="heart"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
        <span>
          Favori
        </span>
      </button>
    </StyledComponent>
  </styled.button>
</NavigationButton>
`;
